openedx/core/djangoapps/user_api/learner_skill_levels/utils.py

```diff
@@ -37,13 +37,17 @@ def generate_skill_score_mapping(user):
 
     skill_score_mapping = {}
     for course_run_id in course_run_ids:
-        course_data = get_course_data(course_run_id, ['skill_names', 'level_type'])
+        # fetch course details from course run id to get course key
+        course_run_data = get_course_run_data(course_run_id, ['course'])
```

Comment:
looks like `course_run_data` can be `None`. 
Do we need to handle it?

Reply:
since we are getting course run ids from `PersistanceCourseGrade` table, it means the course run exists in our system. I don't think the `course_run_data` can ever be null.